tsfile/src/main/java/org/apache/iotdb/tsfile/fileSystem/IoTDBFile.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.tsfile.fileSystem;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.*;
+import org.apache.iotdb.tsfile.common.conf.TSFileConfig;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+
+public class IoTDBFile extends File {
+
+  private Path hdfsPath;
+
+  public IoTDBFile(String pathname) {
+    super(pathname);
+    if (TSFileConfig.isHdfsStorage) {
+      hdfsPath = new Path(pathname);
+    }
+  }
+
+  public IoTDBFile(String parent, String child) {
+    super(parent, child);
+  }
+
+  public IoTDBFile(File parent, String child) {
+    super(parent, child);
+  }
+
+  public IoTDBFile(URI uri) {
+    super(uri);
+  }
+
+  @Override
+  public String getAbsolutePath() {
```

Comment:
Consider another way: Use a IoTDBFileFactory to generate different files: java.io.File and HDFSFile(extends java.io.File). Then, we only need to judge the storage type when create and no need to judge in each method.